plugins/GSdx/GSLzma.cpp

```diff
@@ -162,19 +162,17 @@ bool GSDumpRaw::IsEof() {
 	return feof(m_fp);
 }
 
-void GSDumpRaw::Read(void* ptr, size_t size) {
-	if (size == 1) {
-		// I don't know why but read of size 1 is not happy
-		int v = fgetc(m_fp);
-		memcpy(ptr, &v, 1);
-	} else {
-		size_t ret = fread(ptr, 1, size, m_fp);
-		if (ret != size) {
-			fprintf(stderr, "GSDumpRaw:: Read error (%zu/%zu)\n", ret, size);
-			throw "BAD"; // Just exit the program
-		}
+bool GSDumpRaw::Read(void* ptr, size_t size) {
+	size_t ret = fread(ptr, 1, size, m_fp);
+	if (ret != size && ferror(m_fp)) {
```

Comment:
It makes more sense to use ``||`` over here. What about cases when only one of the condition is false? you won't be throwing the error.

I'm not sure at which scenarios ``ferror()`` returns 1, [reference at some sites](http://www.cplusplus.com/reference/cstdio/ferror/) just mentions that 1 will be set when the previous file operation has failed but it's still not fully clear on how they define a failed operation. Plus [C library code for ferror()](https://sourceware.org/git/?p=glibc.git;a=blob;f=libio/ferror.c;h=90a3b6949625e3b6e913c8705434872a9a8fb94b;hb=9d46370ca338054cb6ea7ebeddcf06c7ac7ad1a9) doesn't make sense to me.